pdc/apps/component/tests.py

```diff
@@ -2508,3 +2509,231 @@ def test_delete_group(self):
 
         response = self.client.get(url, format='json')
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+
+class GlobalComponentContactInfoRESTTestCase(TestCaseWithChangeSetMixin, APITestCase):
+    fixtures = [
+        "pdc/apps/component/fixtures/tests/global_component.json",
+        "pdc/apps/contact/fixtures/tests/contact_role.json",
+        "pdc/apps/contact/fixtures/tests/maillist.json",
+        "pdc/apps/contact/fixtures/tests/person.json",
+        "pdc/apps/component/fixtures/tests/upstream.json"
+    ]
+
+    def setUp(self):
+        super(GlobalComponentContactInfoRESTTestCase, self).setUp()
+        person = Person.objects.get(username='person1')
+        maillist = Maillist.objects.get(mail_name="maillist2")
+        pm_type = ContactRole.objects.get(name='pm')
+        qe_type = ContactRole.objects.get(name='qe_ack')
+
+        python = models.GlobalComponent.objects.get(name='MySQL-python')
+        java = models.GlobalComponent.objects.get(name='java')
+
+        GlobalComponentRoleContact.objects.create(component=python, contact_role=pm_type, contact=person)
+        GlobalComponentRoleContact.objects.create(component=java, contact_role=qe_type, contact=maillist)
+
+    def test_list_global_component_contacts(self):
+        url = reverse('globalcomponentcontacts-list')
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data.get('count'), 2)
+        results = response.data.get('results')
+        self.assertEqual(results[0]['role'], 'pm')
+        self.assertEqual(results[0]['contact']['email'], 'person1@test.com')
+
+        self.assertEqual(results[1]['role'], 'qe_ack')
+        self.assertEqual(results[1]['contact']['mail_name'], 'maillist2')
+
+    def test_retrieve_global_component_contacts(self):
+        url = reverse('globalcomponentcontacts-list')
+        response = self.client.get(url, format='json')
+        results = response.data.get('results')
+        id = results[1]['id']
+        url = reverse('globalcomponentcontacts-detail', kwargs={'pk': id})
+        response = self.client.get(url, format='json')
+        result = response.data
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(result['role'], 'qe_ack')
+        self.assertEqual(result['contact']['mail_name'], 'maillist2')
+        self.assertEqual(result['contact']['email'], 'maillist2@test.com')
+
+    def test_destroy_global_component_contacts(self):
+        url = reverse('globalcomponentcontacts-list')
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.data.get('count'), 2)
+        results = response.data.get('results')
+        id = results[1]['id']
+        url = reverse('globalcomponentcontacts-detail', kwargs={'pk': id})
+        self.client.delete(url, format='json')
+        url = reverse('globalcomponentcontacts-list')
```

Comment:
In 66da6e8a7a17 use changeset to check patch method. It think maybe test create and destroy methods by checking count also straightforward and more concrete.